components/ws-daemon/pkg/cpulimit/cpulimit.go

```diff
@@ -206,7 +219,7 @@ func (d *Distributor) Reset() {
 
 // ResourceLimiter implements a strategy to limit the resurce use of a workspace
 type ResourceLimiter interface {
-	Limit(budgetSpent CPUTime) (newLimit Bandwidth)
+	Limit(wsh *WorkspaceHistory) (Bandwidth, error)
```

Comment:
I feel we must return some value without error because if the error is returned, we can't set any value, in other words, a workspace is unlimited.
```suggestion
	Limit(wsh *WorkspaceHistory) Bandwidth
```

Reply:
This is now taken care of by using the composite limiter which has a fallback that always returns a value (if cpu limiting is enabled and configured, but that is the same behavior as before)